### PR TITLE
Add support for Jackson afterburner module

### PR DESCRIPTION
**Pull request by cemcayiroglu**
Support for Jackson afterburner is added.

### CODE REVIEW

**Comment by cemcayiroglu** [comment]
I ran multiple 2-3 hours tests with production workload on production size clusters. 

CPU improvements on the coordinator:  

Operation | Delta
------------ | -------------
io/airlift/json/JsonCodec.toJsonBytes | Dropped From 32% of total CPU to 15%  of total CPU
io/airlift/json/JsonCodec.fromJson | Stayed same 7% of total CPU.

**Comment by mbasmanova** [comment]
@cemcayiroglu Cem, the performance improvement is quite impressive. Thank's for making this change. I'm going to review it, but I noticed that commit message starts with `Adding` instead of `Add`. Would you fix it?

`Add support for Jackson afterburner module` or `Optimize JSON serde with Jackson afterburner`

File: presto-json/src/main/java/com/facebook/presto/json/PrestoJsonModule.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import com.google.inject.Scopes;
+import io.airlift.json.JsonCodecFactory;
+import io.airlift.json.ObjectMapperProvider;
+
+import static io.airlift.json.JsonBinder.jsonBinder;
+
+public class PrestoJsonModule
+        implements Module
+{
+    private static final String JSON_MODULE_AFTERBURNER_ENABLED = "json.module.after-burner.enabled";
```
Comment (mbasmanova) [inline presto-json/src/main/java/com/facebook/presto/json/PrestoJsonModule.java]: Would you mention this new property in the commit message?

File: presto-json/src/test/java/com/facebook/presto/json/Person.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+public class Person
+{
+    private String name;
+    private boolean rocks;
+    private Optional<String> lastName;
+
+    public static void validatePersonJsonCodec(JsonCodec<Person> jsonCodec)
+    {
+        // create object with null lastName
+        Person expected = new Person().setName("dain").setRocks(true);
```
Comment (mbasmanova) [inline presto-json/src/test/java/com/facebook/presto/json/Person.java]: Any particular reason for choosing this name? Why not John (with capital J)?

File: presto-json/src/test/java/com/facebook/presto/json/Person.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        assertEquals(jsonCodec.fromJson(json), expected);
+
+        bytes = jsonCodec.toJsonBytes(expected);
+        assertEquals(jsonCodec.fromJson(bytes), expected);
+    }
+
+    public static void validatePersonListJsonCodec(JsonCodec<List<Person>> jsonCodec)
+    {
+        ImmutableList<Person> expected = ImmutableList.of(
+                new Person().setName("dain").setRocks(true),
```
Comment (mbasmanova) [inline presto-json/src/test/java/com/facebook/presto/json/Person.java]: hmm... I'd use common English names starting with capital letters: https://www.google.com/search?q=common+english+names

File: presto-json/src/test/java/com/facebook/presto/json/Person.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+                new Person().setName("martin").setRocks(true),
```
Comment (mbasmanova) [inline presto-json/src/test/java/com/facebook/presto/json/Person.java]: Would it make sense to have one entry with rocks == false, one more with empty last name and one more with present last name?

File: presto-json/src/test/java/com/facebook/presto/json/Person.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    {
+        if (this == obj) {
+            return true;
+        }
+        if ((obj == null) || (getClass() != obj.getClass())) {
+            return false;
+        }
+        Person o = (Person) obj;
+        return Objects.equals(this.name, o.name) &&
+                Objects.equals(this.rocks, o.rocks);
```
Comment (mbasmanova) [inline presto-json/src/test/java/com/facebook/presto/json/Person.java]: lastName is missing?

File: presto-json/src/test/java/com/facebook/presto/json/Person.java [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(name, rocks);
```
Comment (mbasmanova) [inline presto-json/src/test/java/com/facebook/presto/json/Person.java]: last name?

File: presto-json/src/test/java/com/facebook/presto/json/Person.java (hunk 2 above)
Comment (cemcayiroglu) [reply presto-json/src/test/java/com/facebook/presto/json/Person.java]: I copied the code from Airlift :) I am going to clean this part.

**Comment by cemcayiroglu** [comment]
@mbasmanova I addressed your comments.

**Comment by cemcayiroglu** [comment]
I am closing this since it is handled here: 12366

**Comment by cemcayiroglu** [comment]
https://github.com/prestodb/presto/pull/12366

**Pull request closed without merging**